apps/omg_child_chain/lib/omg_child_chain/fees/feed_adapter.ex

```diff
@@ -0,0 +1,55 @@
+# Copyright 2019-2020 OmiseGO Pte Ltd
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+defmodule OMG.ChildChain.Fees.FeedAdapter do
+  @moduledoc """
+  Adapter pulls actual fees prices from fee feed.
+  """
+  @behaviour OMG.ChildChain.Fees.Adapter
+
+  alias OMG.ChildChain.Fees.FeeUpdater
+  alias OMG.ChildChain.HttpRPC.Client
+  use OMG.Utils.LoggerExt
+
+  @doc """
+  Pulls the fee specification from fees feed. Feed updates fee prices based on Ethereum's gas price.
+  """
+  # sobelow_skip ["Traversal"]
+  @impl true
+  def get_fee_specs(actual_fee_specs, updated_at) do
+    opts = Application.fetch_env!(:omg_child_chain, :fee_adapter_opts)
+    feed_url = Keyword.fetch!(opts, :feed_url)
+
+    with {:ok, fee_specs_from_feed} <- Client.all_fees(feed_url),
```

Comment:
Are you suggesting that we should let the FeeServer process crash and rely on supervisor to bring it back?
I've seen it already in a few places.
In this case I doubt because I'm providing an adapter that should align to the adapter behaviour, and let the FeeServer to handle the error
https://github.com/omisego/elixir-omg/blob/bf6670cf27f1baca92df6be68edc84a576e83738/apps/omg_child_chain/lib/omg_child_chain/fee_server.ex#L118

WDYT?

Reply:
It seems like the feeserver is written too defensively, because, whatever the fee specs are, if they're erroneous, the process should fail and provide a stacktrace. 

that is, if this these fees specs are dropped and are not going to bring the entire application down because they would be repeated.

Reply:
think about it this way: if you're able to avoid these fees with a crash, you should crash. if you cannot avoid erroneous fees being polled, you log the error

Reply:
Primo, I cannot avoid any network error that can happen.
Secundo, as an adapter I let the caller (`FeeServer`) deal with errors. It can crash, log, raise alerts...

Reply:
yeah, network errors are transient, that's why you let the stateful process handle it and crash
on the secundo... you're not doing that, you're just logging